### ISSUE TITLE
AIX failures

### ISSUE THREAD

**Issue by karel-m** [libtom/libtomcrypt#209]
While working on #208 I have spent some time with esoteric compilers from IBM and HP.

Here are some maybe interesting warnings/errors I have experienced with IBM C compiler:

```
"src/modes/xts/xts_test.c", line 28.9: 1506-068 (W) Operation between types "void*" and "int(*)(const unsigned char*,unsigned char*,unsigned long,unsigned char*,union Symmetric_key*,union Symmetric_key*)" is not allowed.
"src/modes/xts/xts_test.c", line 35.52: 1506-068 (W) Operation between types "int(*)(const unsigned char*,unsigned char*,unsigned long,unsigned char*,union Symmetric_key*,union Symmetric_key*)" and "void*" is not allowed.
"src/modes/xts/xts_test.c", line 53.9: 1506-068 (W) Operation between types "void*" and "int(*)(const unsigned char*,unsigned char*,unsigned long,unsigned char*,union Symmetric_key*,union Symmetric_key*)" is not allowed.
"src/modes/xts/xts_test.c", line 60.52: 1506-068 (W) Operation between types "int(*)(const unsigned char*,unsigned char*,unsigned long,unsigned char*,union Symmetric_key*,union Symmetric_key*)" and "void*" is not allowed.

"src/pk/dh/dh_static.h", line 73.1: 1506-277 (S) Syntax error: possible missing ';' or ','?
"src/pk/dh/dh_static.h", line 72.8: 1506-485 (S) Parameter declaration list is incompatible with declarator for inline.
"src/pk/dh/dh_static.h", line 90.15: 1506-045 (S) Undeclared identifier src.
"src/pk/dh/dh_static.h", line 99.8: 1506-045 (S) Undeclared identifier section.
"src/pk/dh/dh_static.h", line 99.34: 1506-045 (S) Undeclared identifier subsection.
```

**Comment by sjaeckel** [comment]
> ```
> "src/modes/xts/xts_test.c", line 28.9: 1506-068 (W) Operation between types "void*" and "int(*)(const unsigned char*,unsigned char*,unsigned long,unsigned char*,union Symmetric_key*,union Symmetric_key*)" is not allowed.
> ...
> ```

very true

> ```
> "src/pk/dh/dh_static.h", line 73.1: 1506-277 (S) Syntax error: possible missing ';' or ','?
> ...
> ```

ugh, that compiler seems to do fancy stuff for inline functions...

**Comment by karel-m** [comment]
BTW HP cc compiler complains about `xts_test.c` like this:
```
cc: "src/modes/xts/xts_test.c", line 28: warning 604: Pointers are not assignment-compatible.
cc: "src/modes/xts/xts_test.c", line 35: warning 604: Pointers are not assignment-compatible.
cc: "src/modes/xts/xts_test.c", line 53: warning 604: Pointers are not assignment-compatible.
cc: "src/modes/xts/xts_test.c", line 60: warning 604: Pointers are not assignment-compatible.
```
Do we want to fix `xts_test.c`?

**Comment by sjaeckel** [comment]
> Do we want to fix xts_test.c?

```
diff --git a/src/modes/xts/xts_test.c b/src/modes/xts/xts_test.c
index 0b887ad..7cbd3be 100644
--- a/src/modes/xts/xts_test.c
+++ b/src/modes/xts/xts_test.c
@@ -19,3 +19,5 @@ static int _xts_test_accel_xts_encrypt(const unsigned char *pt, unsigned char *c
    symmetric_xts xts;
-   void *orig;
+   int (*orig)(const unsigned char *, unsigned char *,
+               unsigned long , unsigned char *, symmetric_key *,
+               symmetric_key *);
 
@@ -44,3 +46,5 @@ static int _xts_test_accel_xts_decrypt(const unsigned char *ct, unsigned char *p
    symmetric_xts xts;
-   void *orig;
+   int (*orig)(const unsigned char *, unsigned char *,
+                unsigned long , unsigned char *, symmetric_key *,
+                symmetric_key *);
 
```
yes, please add the fix to #210

**Comment by karel-m** [comment]
added, I am closing this issue, let's move to #210 with further comments
